homeassistant/components/sunsynkweb/sensor.py

```diff
@@ -0,0 +1,177 @@
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        key="acc_pv",
+        translation_key="acc_pv",
+        suggested_display_precision=2,
+    ),
+    SunsynkEntitySensorDescription(  # ""Total energy consumed through the inverters."""
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        key="acc_load",
+        translation_key="acc_load",
+        suggested_display_precision=2,
+    ),
+    SunsynkEntitySensorDescription(  # ""Total energy imported from the grid."""
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        key="acc_grid_import",
+        translation_key="acc_grid_import",
+        suggested_display_precision=2,
+        agg_func=max,
+    ),
+    SunsynkEntitySensorDescription(  # ""Total energy exported to the grid."""
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        key="acc_grid_export",
+        translation_key="acc_grid_export",
+        suggested_display_precision=2,
+        agg_func=max,
+    ),
+    SunsynkEntitySensorDescription(  # ""Total energy injected into the batteries."""
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        key="acc_battery_charge",
+        translation_key="acc_battery_charge",
+        suggested_display_precision=2,
+    ),
+    SunsynkEntitySensorDescription(  # ""Total energy provided by the batteries."""
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        key="acc_battery_discharge",
+        translation_key="acc_battery_discharge",
+        suggested_display_precision=2,
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: SunsynkConfigEntry,
+    async_add_entities: AddEntitiesCallback,  # noqa: F821
+) -> None:
+    """Set up sensor devices."""
+    coordinator: SunsynkUpdateCoordinator = entry.runtime_data
+
+    async_add_entities(
+        SunSynkApiSensor(coordinator, description)
+        for description in SENSOR_DESCRIPTIONS
+    )
+
+
+class SunSynkApiSensor(CoordinatorEntity[SunsynkUpdateCoordinator], SensorEntity):
+    """Parent class for sunsynk api exposing sensors.
+
+    The sensors expose the sum of the power across all inverters
+    for plants that have more than one inverter.
+    State of charge is normally shared between inverters, so this will take
+    the maximum of the state of charge.
+    """
+
+    _attr_has_entity_name = True
+
+    @property
+    def device_info(self) -> DeviceInfo:
```

Comment:
can you use the _attr instead (it's just a bit shorter). Also can we add the manufacturer?

Reply:
done